Civi/Payment/PropertyBag.php

```diff
@@ -174,13 +184,16 @@ public function offsetGet($offset) {
       }
     }
 
+    // These lines are here (and not in try block) because the catch must only
+    // catch the case when the prop is custom.
+    $getter = 'get' . ucfirst($prop);
     if (!$this->getSuppressLegacyWarnings()) {
       CRM_Core_Error::deprecatedFunctionWarning(
         "get" . ucfirst($offset) . "()",
         "PropertyBag array access for core property '$offset'"
       );
     }
-    return $this->get($prop, 'default');
+    return $this->$getter('default');
```

Comment:
Ensure that property "getter" is used instead of generic getter (so we don't skip special handling eg. default for getIsRecur()).

Reply:
Merged via #21583